operator/src/test/java/org/keycloak/operator/RealmImportE2EIT.java

```diff
@@ -0,0 +1,153 @@
+                        .withNamespace(namespace)
+                        .endMetadata()
+                        .withNewSpec()
+                        .withSelector(Map.of("app", "keycloak"))
+                        .addNewPort()
+                        .withPort(KEYCLOAK_PORT)
+                        .endPort()
+                        .endSpec()
+                        .build()
+        );
+
+        // Act
+        k8sclient.load(getClass().getResourceAsStream("/example-realm.yaml")).inNamespace(namespace).createOrReplace();
+
+        // Assert
+        Awaitility.await()
+                .atMost(3, MINUTES)
+                .pollDelay(5, SECONDS)
+                .ignoreExceptions()
+                .untilAsserted(() -> {
+                    var conditions = k8sclient
+                            .resources(KeycloakRealmImport.class)
+                            .inNamespace(namespace)
+                            .withName("example-count0-kc")
+                            .get()
+                            .getStatus()
+                            .getConditions();
+
+                    assertThat(getCondition(conditions, DONE).getStatus()).isTrue();
+                    assertThat(getCondition(conditions, STARTED).getStatus()).isFalse();
+                    assertThat(getCondition(conditions, HAS_ERRORS).getStatus()).isFalse();
+                });
+
+        String url =
+                "http://" + KEYCLOAK_SERVICE_NAME + "." + namespace + ":" + KEYCLOAK_PORT + "/realms/count0";
+
+        Awaitility.await().atMost(5, MINUTES).untilAsserted(() -> {
+            try {
+                Log.info("Starting curl Pod to test if the realm is available");
+
+                Pod curlPod = k8sclient.run().inNamespace(namespace)
+                        .withRunConfig(new RunConfigBuilder()
+                                .withArgs("-s", "-o", "/dev/null", "-w", "%{http_code}", url)
+                                .withName("curl")
+                                .withImage("curlimages/curl:7.78.0")
+                                .withRestartPolicy("Never")
+                                .build())
+                        .done();
+                Log.info("Waiting for curl Pod to finish running");
+                Awaitility.await().atMost(2, MINUTES)
+                        .until(() -> {
+                            String phase =
+                                    k8sclient.pods().inNamespace(namespace).withName("curl").get()
+                                            .getStatus().getPhase();
+                            return phase.equals("Succeeded") || phase.equals("Failed");
+                        });
+
+                String curlOutput =
+                        k8sclient.pods().inNamespace(namespace)
+                                .withName(curlPod.getMetadata().getName()).getLog();
+                Log.info("Output from curl: '" + curlOutput + "'");
+                assertThat(curlOutput).isEqualTo("200");
```

Comment:
This is a neat trick to avoid the hassle with accessing KC from the tests (that run outside of the cluster). Normally I'd ask to make it more generic as this logic could come in handy in other tests too. But we still need to figure out testing Ingress as part of #9731, so we can rewrite this later to use Ingress.

Reply:
@andreaTP yep, feel free to do so :)